apps/04TowersOfHanoi.js

```diff
@@ -16,25 +16,75 @@ function printStacks() {
     console.log("c: " + stacks.c);
 }
 
-function movePiece() {
-    // Your code here
+function movePiece(startStack, endStack) {
+
+  var piece = stacks[startStack].pop()
+  stacks[endStack].push(piece);
 
 }
 
-function isLegal() {
-    // Your code here
+function isLegal(startStack, endStack) {
 
-}
+
+  var startStackArray = stacks[startStack];
+
+  var endStackArray = stacks[endStack];
+
+  if ( startStackArray.length === 0 )   {
+
+    return false;
+
+  }
+
+  else if (endStackArray.length === 0)  {
+
+      return true;
+
+  }
+
+  else {
+
+      return startStackArray[startStackArray.length - 1] < endStackArray[endStackArray.length - 1];
+
+  }
+
+  }
 
 function checkForWin() {
     // Your code here
+    if (stacks.c.length === 4)   {
+
+    console.log("You Won!!")
+
+    return true
+
+    }
+
+    if (stacks.b.length === 4)    {
+
+      console.log("You Won!!")
+
+      return true
+
+    }
+
+    else {
+
+      return false
+
+    }
 
 }
 
 function towersOfHanoi(startStack, endStack) {
-    // Your code here
 
-}
+
+
+// Your code here
+
+  movePiece(startStack, endStack);
```

Comment:
This looks a little off to me. Do we always move the piece? Why'd we write that isLegal function?